README.md
[![Release](https://img.shields.io/badge/Version-^0.2.1-success.svg?style=for-the-badge)](https://shields.io/)
[![Maintained](https://img.shields.io/badge/Maintained-Actively-informational.svg?style=for-the-badge)](https://shields.io/)

# shared_themes

`shared_themes` is an open source project &mdash; it's one among many other shared libraries that make up the wider ecosystem of software made and open sourced by `Savannah Informatics Limited`.

A shared library for `BeWell-Consumer` and `SladeAdvantage` that is responsible for defining and providing theme/style guidelines

## Installation Instructions

Use this package as a library by depending on it

Run this command:

- With Flutter:

```dart
$ flutter pub add shared_themes
```

This will add a line like this to your package's pubspec.yaml (and run an implicit dart pub get):

```dart
dependencies:
  shared_themes: ^0.2.1
```

Alternatively, your editor might support flutter pub get. Check the docs for your editor to learn more.

Lastly:

Import it like so:

```dart
import 'package:shared_themes/colors.dart';
```

## Usage

Check the [example](https://github.com/savannahghi/shared_themes/blob/main/example/main.dart) provided for how to use this package.

## Dart & Flutter Version

- Dart 2: >= 2.12
- Flutter: >=2.0.0

## Developing & Contributing

First off, thanks for taking the time to contribute!

Be sure to check out detailed instructions on how to contribute to this project [here](https://github.com/savannahghi/shared_themes/blob/main/CONTRIBUTING.md) and go through out [Code of Conduct](https://github.com/savannahghi/shared_themes/blob/main/CODE_OF_CONDUCT.md).

GPG Signing: 
As a contributor, you need to sign your commits. For more details check [here](https://docs.github.com/en/github/authenticating-to-github/managing-commit-signature-verification/signing-commits)

## License

This library is distributed under the MIT license found in the [LICENSE](https://github.com/savannahghi/shared_themes/blob/main/LICENSE) file.

CHANGELOG.md
### 0.2.1

- chore: upgrade to flutter 3.3.8
- upgrade dependencies

### 0.2.0

- chore: upgrade to flutter 3.0.5
- feat: setup automated deployment to pub.dev

### 0.1.4

- refactor: add color

### 0.1.3

- refactor: update colors

### 0.1.2

- refactor: support phone number string

### 0.1.1

- chore: add new colors

### 0.1.0

- doc: project documentation fixes

### 0.0.9

- chore: update dependencies (google_fonts, lint, )

### Pre - [0.0.8]

- Code testing, refactor and upgrading design system for shared_themes on both apps
- Assorted bug fixes and enhancements
- Migrated to nullsafety


test/colors_test.dart
import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';
import 'package:shared_themes/colors.dart';

void main() {
  test('color gradient renders correctly', () {
    final List<Color> silLinearGradientColors = <Color>[
      healthcloudTextColorPurple,
      healthcloudPrimaryColor,
    ];
    expect(
      silLinearGradientColors,
      const <Color>[Color(0xFF6A54B7), Color(0xFF7949AF)],
    );
  });
  test('should test background gradient', () {
    final List<Color> gradient = silLinearGradientColors;
    const Color grey = greyCardBackground;
    const Color background = appBackgroundColor;
    const Color containerColor = containerBackgroundColor;

    expect(containerColor, const Color(0xffF6F7FB));

    expect(background, const Color(0xffF4F5FA));

    expect(grey, const Color(0xfff7f7f9));

    expect(gradient, isA<List<Color>>());
  });
}


test/constants_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:shared_themes/constants.dart';

void main() {
  test('getSuccessfulAddMessage', () {
    const String process = 'Cover added';
    final String message = UserFeedBackTexts.getSuccessfulAddMessage(process);
    expect(message, '$process added successfully');
  });

  test('getSuccessfulRetireMessage', () {
    const String process = 'Retire';
    final String message =
        UserFeedBackTexts.getSuccessfulRetireMessage(process);
    expect(message, '$process retired successfully');
  });

  test('getErrorMessage ', () {
    const String process = 'processing';
    final String message = UserFeedBackTexts.getErrorMessage(process);
    expect(
        message,
        'Sorry, an error occurred while $process. Please try again later, '
        'or contact Slade 360 Be.Well Support on $kBewellSupportPhoneNumber');
  });

  test('getErrorMessage when isEmpty', () {
    final String message = UserFeedBackTexts.getErrorMessage();
    expect(
        message,
        'Sorry, an error occurred. Please try again, or contact '
        'Slade 360 Be.Well Support support on $kBewellSupportPhoneNumber');
  });

// This value is subject to discussion
  test('should return ios versions', () {
    expect(supportedIosVersions, <String>['13', '14', '15', '16', '17', '18']);
  });

  test('should show the Kenyan and americanPhoneRegExp regex exists', () {
    expect(
      kenyanPhoneRegExp,
      RegExp(
        r'''^(?:254|\+254|0)?((7|1)(?:(?:[0-9][0-9])|(?:0[0-8])|(4[0-1]))[0-9]{6})$''',
      ),
    );
    expect(
      americanPhoneRegExp,
      RegExp(r'''^(\+)(\d{1,})([(]{1}\d{1,3}[)]){0,}\d{2}\d{3}\d{5}$'''),
    );

    expect(
      unitedKingdomRegExp,
      RegExp(
        r'''^(?:(?:\(?(?:0(?:0|11)\)?[\s-]?\(?|\+)44\)?[\s-]?(?:\(?0\)?[\s-]?)?)|(?:\(?0))(?:(?:\d{5}\)?[\s-]?\d{4,5})|(?:\d{4}\)?[\s-]?(?:\d{5}|\d{3}[\s-]?\d{3}))|(?:\d{3}\)?[\s-]?\d{3}[\s-]?\d{3,4})|(?:\d{2}\)?[\s-]?\d{4}[\s-]?\d{4}))(?:[\s-]?(?:x|ext\.?|\#)\d{3,4})?$''',
      ),
    );

    expect(
      genericInternationalRegExp,
      RegExp(
        r'''\+(9[976]\d|8[987530]\d|6[987]\d|5[90]\d|42\d|3[875]\d|2[98654321]\d|9[8543210]|8[6421]|6[6543210]|5[87654321]|4[987654310]|3[9643210]|2[70]|7|1)\d{1,14}$''',
      ),
    );
  });

  test('should return true if number is Kenyan', () {
    const String testKenyanNumber1 = '0735000000';
    const String testKenyanNumber2 = '0720000000';
    const String testKenyanNumber3 = '0790000000';
    const String testKenyanNumber4 = '0110000000';

    final bool result1 = kenyanPhoneRegExp.hasMatch(testKenyanNumber1);
    final bool result2 = kenyanPhoneRegExp.hasMatch(testKenyanNumber2);
    final bool result3 = kenyanPhoneRegExp.hasMatch(testKenyanNumber3);
    final bool result4 = kenyanPhoneRegExp.hasMatch(testKenyanNumber4);

    expect(result1, true);
    expect(result2, true);
    expect(result3, true);
    expect(result4, true);
  });

  test('should show the email regex exists', () {
    expect(
      emailValidator,
      RegExp(
        r"^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,253}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,253}[a-zA-Z0-9])?)*$",
      ),
    );
  });

  test('selectOrTakeMessage', () {
    const String name = 'your ID';
    final String message = UserFeedBackTexts.selectOrTakeMessage(name);
    expect(message, 'Select or take a photo of $name');
  });
}


test/text_theme_test.dart
import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';
import 'package:google_fonts/google_fonts.dart';
import 'package:shared_themes/text_themes.dart';

void main() {
  test('should expect the right text styles for Text Theme', () {
    Color? color;

//size 50
    final TextStyle heavySize50Text = TextThemes.heavySize50Text();
    expect(
      heavySize50Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w700,
        color: color,
        fontSize: 55,
      ),
    );
    // size 45
    final TextStyle boldSize45Text = TextThemes.boldSize45Text();
    expect(
      boldSize45Text,
      GoogleFonts.montserrat(
        fontSize: 45,
        color: color,
        fontWeight: FontWeight.w500,
      ),
    );

    // size 34
    final TextStyle normalSize34Text = TextThemes.normalSize34Text();
    expect(
      normalSize34Text,
      GoogleFonts.montserrat(fontSize: 34, color: color),
    );

    //size 32
    final TextStyle veryBoldSize32Text = TextThemes.veryBoldSize32Text();
    expect(
      veryBoldSize32Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w600,
        color: color,
        fontSize: 32,
      ),
    );

    final TextStyle normalSize32Text = TextThemes.normalSize32Text();
    expect(
      normalSize32Text,
      GoogleFonts.montserrat(fontSize: 32, color: color),
    );

    //size 30
    final TextStyle boldSize30Text = TextThemes.boldSize30Text();
    expect(
      boldSize30Text,
      GoogleFonts.montserrat(
        fontSize: 30,
        fontWeight: FontWeight.w500,
        color: color,
      ),
    );

    //size 28
    final TextStyle veryBoldSize28Text = TextThemes.veryBoldSize28Text();
    expect(
      veryBoldSize28Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w600,
        color: color,
        fontSize: 28,
      ),
    );

    final TextStyle heavySize28Text = TextThemes.heavySize28Text();
    expect(
      heavySize28Text,
      GoogleFonts.montserrat(
        color: color,
        fontWeight: FontWeight.w700,
        fontSize: 28,
      ),
    );

    final TextStyle normalSize28Text = TextThemes.normalSize28Text();
    expect(
      normalSize28Text,
      GoogleFonts.montserrat(fontSize: 28, color: color),
    );

    //size 26

    final TextStyle heavySize26Text = TextThemes.heavySize26Text();
    expect(
      heavySize26Text,
      GoogleFonts.montserrat(
        color: color,
        fontWeight: FontWeight.w700,
        fontSize: 26,
      ),
    );

    final TextStyle normalSize26Text = TextThemes.normalSize26Text();
    expect(
      normalSize26Text,
      GoogleFonts.montserrat(fontSize: 26, color: color),
    );

    final TextStyle boldSize26Title = TextThemes.boldSize26Title();
    expect(
      boldSize26Title,
      GoogleFonts.montserrat(
        fontSize: 26,
        color: color,
        fontWeight: FontWeight.w500,
      ),
    );

    // size 24
    final TextStyle heavySize24Text = TextThemes.heavySize24Text();
    expect(
      heavySize24Text,
      GoogleFonts.montserrat(
        color: color,
        fontWeight: FontWeight.w700,
        fontSize: 24,
      ),
    );
    final TextStyle mediumSize24Text = TextThemes.mediumSize24Text();
    expect(
      mediumSize24Text,
      GoogleFonts.montserrat(
        color: color,
        fontWeight: FontWeight.w600,
        fontSize: 24,
      ),
    );
    final TextStyle boldSize24Text = TextThemes.boldSize24Text();
    expect(
      boldSize24Text,
      GoogleFonts.montserrat(
        color: color,
        fontWeight: FontWeight.w500,
        fontSize: 24,
      ),
    );
    final TextStyle normalSize24Text = TextThemes.normalSize24Text();
    expect(
      normalSize24Text,
      GoogleFonts.montserrat(color: color, fontSize: 24),
    );

    final TextStyle veryBoldSize22Text = TextThemes.veryBoldSize22Text();
    expect(
      veryBoldSize22Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w600,
        color: color,
        fontSize: 22,
      ),
    );
    final TextStyle boldSize22Text = TextThemes.boldSize22Text();
    expect(
      boldSize22Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w500,
        color: color,
        fontSize: 22,
      ),
    );

    final TextStyle mediumSize22Text = TextThemes.mediumSize22Text();
    expect(
      mediumSize22Text,
      GoogleFonts.montserrat(
        color: color,
        fontWeight: FontWeight.w500,
        fontSize: 22,
      ),
    );

    final TextStyle normalSize22Text = TextThemes.normalSize22Text();
    expect(
      normalSize22Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w400,
        color: color,
        fontSize: 22,
      ),
    );

    // size 20
    final TextStyle veryHeavySize20Text = TextThemes.veryHeavySize20Text();
    expect(
      veryHeavySize20Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w800,
        color: color,
        fontSize: 20,
      ),
    );
    final TextStyle heavySize20Text = TextThemes.heavySize20Text();
    expect(
      heavySize20Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w700,
        color: color,
        fontSize: 20,
      ),
    );
    final TextStyle veryBoldSize20Text = TextThemes.veryBoldSize20Text();
    expect(
      veryBoldSize20Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w600,
        color: color,
        fontSize: 20,
      ),
    );
    final TextStyle boldSize20Text = TextThemes.boldSize20Text();
    expect(
      boldSize20Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w500,
        color: color,
        fontSize: 20,
      ),
    );

    final TextStyle mediumSize20Text = TextThemes.mediumSize20Text();
    expect(
      mediumSize20Text,
      GoogleFonts.montserrat(
        color: color,
        fontWeight: FontWeight.w500,
        fontSize: 20,
      ),
    );

    final TextStyle normalSize20Text = TextThemes.normalSize20Text();
    expect(
      normalSize20Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w400,
        color: color,
        fontSize: 20,
      ),
    );

    // size 18
    final TextStyle heavySize18Text = TextThemes.heavySize18Text();
    expect(
      heavySize18Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w700,
        color: color,
        fontSize: 18,
      ),
    );
    final TextStyle veryBoldSize18Text = TextThemes.veryBoldSize18Text();
    expect(
      veryBoldSize18Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w600,
        color: color,
        fontSize: 18,
      ),
    );
    final TextStyle boldSize18Text = TextThemes.boldSize18Text();
    expect(
      boldSize18Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w500,
        color: color,
        fontSize: 18,
      ),
    );
    final TextStyle normalSize18Text = TextThemes.normalSize18Text();
    expect(
      normalSize18Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w400,
        color: color,
        fontSize: 18,
      ),
    );

    // size 17
    final TextStyle veryBoldSize17Text = TextThemes.veryBoldSize17Text();
    expect(
      veryBoldSize17Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w600,
        color: color,
        fontSize: 17,
      ),
    );
    final TextStyle normalSize17Text = TextThemes.normalSize17Text();
    expect(
      normalSize17Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w400,
        color: color,
        fontSize: 17,
      ),
    );

    // size 16
    final TextStyle heavySize16Text = TextThemes.heavySize16Text();
    expect(
      heavySize16Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w700,
        color: color,
        fontSize: 16,
      ),
    );
    final TextStyle veryBoldSize16Text = TextThemes.veryBoldSize16Text();
    expect(
      veryBoldSize16Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w600,
        color: color,
        fontSize: 16,
      ),
    );
    final TextStyle boldSize16Text = TextThemes.boldSize16Text();
    expect(
      boldSize16Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w500,
        color: color,
        fontSize: 16,
      ),
    );
    final TextStyle normalSize16Text = TextThemes.normalSize16Text();
    expect(
      normalSize16Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w400,
        color: color,
        fontSize: 16,
      ),
    );
    final TextStyle lightSize16Text = TextThemes.lightSize16Text();
    expect(
      lightSize16Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w300,
        color: color,
        fontSize: 16,
      ),
    );

    // size 15
    final TextStyle veryBoldSize15Text = TextThemes.veryBoldSize15Text();
    expect(
      veryBoldSize15Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w600,
        color: color,
        fontSize: 15,
      ),
    );
    final TextStyle boldSize15Text = TextThemes.boldSize15Text();
    expect(
      boldSize15Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w500,
        color: color,
        fontSize: 15,
      ),
    );
    final TextStyle normalSize15Text = TextThemes.normalSize15Text();
    expect(
      normalSize15Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w400,
        color: color,
        fontSize: 15,
      ),
    );

    // size 14
    final TextStyle veryHeavySize14Text = TextThemes.veryHeavySize14Text();
    expect(
      veryHeavySize14Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w800,
        color: color,
        fontSize: 14,
      ),
    );
    final TextStyle heavySize14Text = TextThemes.heavySize14Text();
    expect(
      heavySize14Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w700,
        color: color,
        fontSize: 14,
      ),
    );
    final TextStyle boldSize14Text = TextThemes.boldSize14Text();
    expect(
      boldSize14Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w500,
        color: color,
        fontSize: 14,
      ),
    );
    final TextStyle veryBoldSize14Text = TextThemes.veryBoldSize14Text();
    expect(
      veryBoldSize14Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w600,
        color: color,
        fontSize: 14,
      ),
    );
    final TextStyle normalSize14Text = TextThemes.normalSize14Text();
    expect(
      normalSize14Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w400,
        color: color,
        fontSize: 14,
      ),
    );

    final TextStyle mediumSize14Text = TextThemes.mediumSize14Text();
    expect(
      mediumSize14Text,
      GoogleFonts.montserrat(
        color: color,
        fontWeight: FontWeight.w600,
        fontSize: 14,
      ),
    );
    // size 13
    final TextStyle boldSize13Text = TextThemes.boldSize13Text();
    expect(
      boldSize13Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w500,
        color: color,
        fontSize: 13,
      ),
    );
    final TextStyle normalSize13Text = TextThemes.normalSize13Text();
    expect(
      normalSize13Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w400,
        color: color,
        fontSize: 13,
      ),
    );

    // size 12
    final TextStyle heavySize12Text = TextThemes.heavySize12Text();
    expect(
      heavySize12Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w700,
        color: color,
        fontSize: 12,
      ),
    );

    final TextStyle veryHeavySize12Text = TextThemes.veryHeavySize12Text();
    expect(
      veryHeavySize12Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w800,
        color: color,
        fontSize: 12,
      ),
    );

    final TextStyle boldSize12Text = TextThemes.boldSize12Text();
    expect(
      boldSize12Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w500,
        color: color,
        fontSize: 12,
      ),
    );
    final TextStyle normalSize12Text = TextThemes.normalSize12Text();
    expect(
      normalSize12Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w400,
        color: color,
        fontSize: 12,
      ),
    );

    // size 11
    final TextStyle heavySize11Text = TextThemes.heavySize11Text();
    expect(
      heavySize11Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w700,
        color: color,
        fontSize: 11,
      ),
    );
    final TextStyle normalSize11Text = TextThemes.normalSize11Text();
    expect(
      normalSize11Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w400,
        color: color,
        fontSize: 11,
      ),
    );

    // size 10
    final TextStyle heavySize10Text = TextThemes.heavySize10Text();
    expect(
      heavySize10Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w700,
        color: color,
        fontSize: 10,
      ),
    );
    final TextStyle normalSize10Text = TextThemes.normalSize10Text();
    expect(
      normalSize10Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w400,
        color: color,
        fontSize: 10,
      ),
    );

    // size 9
    final TextStyle normalSize9Text = TextThemes.normalSize9Text();
    expect(
      normalSize9Text,
      GoogleFonts.montserrat(
        fontWeight: FontWeight.w400,
        color: color,
        fontSize: 9,
      ),
    );
  });
}


lib/colors.dart
import 'package:flutter/material.dart';

const Color sILPurple = Color(0xff7B54C4);

// SIL brave blue
const Color sILBraveBlue = Color(0xff0089ff);

// Cyan used for buttons
const Color sILButtonColor = Colors.cyan;

//Primary color - cyan
const Color healthcloudPrimaryColor = Color(0xFF7949AF);

// accent color - purple
const Color healthcloudAccentColor = Color(0xFF739A00);

//light grey
const Color healthcloudBackgroundColor = Color(0xFFF4F6FA);

//darkblue
const Color healthcloudTextColor = sILBraveBlue;

//purple
const Color healthcloudTextColorPurple = Color(0xFF6A54B7);

const Color patientIDLightGrey = Color(0xFFC4C4C4);

const Color titleBlack = Color(0xFF1B1B1B);

const Color subtitleGrey = Color(0xFF666666);

const Color patientIDResultCardColor = Color(0xFFFCFCFC);

const Color patientIDLightCyan = Color(0xFFCAEAE7);

const Color patientExamDrawerColor = Color(0xfff4f4f4);

const Color darkGreenButtonTextColor = Color(0xff009e91);

const Color lightGreenButtonFillColor = Color(0xffcaeae7);

const Color patientSearchBorderColor = Color(0xfff2f2f2);

const Color avatarBackgroundColor = Color(0xff8fcad6);

const Color titleColor = Color(0xFF1A1A1A);

const Color subtitleColor = Color(0xFFA9A9A9);

const Color textBodyColor = Color(0xFF262626);

const Color white = Colors.white;

const Color redAccent = Colors.redAccent;

const Color grey = Colors.grey;

const Color black = Colors.black;

const Color black54 = Colors.black54;

const Color red = Colors.red;

const Color transparent = Colors.transparent;

const Color green = Colors.green;

/// used in the sign up and verification pages
const Color authBackgroundColor = Color(0xffeff5ff);

/* status colors
/ [successColor] is used to communicate presence of a positive action
/ [cautionColor] is used to communicate a warning
/ [dangerColor] is used to communicate danger or remove buttons
/ [notificationColor] is used to show notifications
*/

const Color cautionColor = Color(0xfff99330);

const Color dangerColor = Color(0xffb00020);

const Color disabledButtonColor = Color(0xffc8c8c8);

/// gradient colors
final List<Color> silLinearGradientColors = <Color>[
  healthcloudTextColorPurple,
  healthcloudPrimaryColor,
];

const Color greyCardBackground = Color(0xfff7f7f9);

const Color appBackgroundColor = Color(0xffF4F5FA);

const Color containerBackgroundColor = Color(0xffF6F7FB);

const Color primaryGreenSelectedColor = Color(0xff144C4A);
const Color secondaryGreenSelectedColor = Color(0xff3D7B78);
const Color tertiaryGreenSelectedColor = Color(0xffB4E7DB);
const Color backgroundGreenSelectedColor = Color(0xffDBF4ED);
const Color purpleBackgroundColor = Color(0xffE8E4F1);
const Color linkColor = Color(0xff448AFF);
const Color subtitleTextColor = Color(0xff7E7D7E);
const Color secondaryTextColor = Color(0xff9077B9);
const Color textFiledGreyColor = Color(0xffC4C4C4);
const Color secondaryNavButtonColor = Color(0xffE0E0E0);
const Color primaryNavButtonColor = Color(0xff53C451);
const Color backgroundGreyColor = Color(0xffF2F2F2);
const Color primaryErrorColor = Color(0xffF85C5D);
const Color mainWarningColor = Color(0xff5C4914);
const Color secondaryWarningColor = Color(0xff94772E);
const Color warningBackgroundColor = Color(0xffFFFCF4);
const Color primaryNudgeColor = Color(0xff90A955);
const Color secondaryNudgeColor = Color(0xff5F99FA);
const Color urgentNudgeColor = Color(0xffED7A7A);
const Color catskillWhiteColor = Color(0xffF9FAFC);


check_coverage.py
#!/usr/bin/python3
import os
import re
import string
import sys

from bs4 import BeautifulSoup


def checkCoverage():
    THRESHOLD = 100.00

    current_dir = os.getcwd()
    coverage_file_path = os.path.join(current_dir, "coverage/index.html")
    cov = None

    with open(coverage_file_path) as file:
        soup = BeautifulSoup(file, "lxml")
        covHi = soup.find(attrs={"class": "headerCovTableEntryHi"})
        covMed = soup.find(attrs={"class": "headerCovTableEntryMed"})
        covLow = soup.find(attrs={"class": "headerCovTableEntryLo"})

        if hasattr(covHi, "string") and covHi.string != "-":
            cov = covHi.string
        elif hasattr(covMed, "string") and covMed.string != "-":
            cov = covMed.string
        elif hasattr(covLow, "string") and covLow.string != "-":
            cov = covLow.string

    pattern = r"[{}]".format(string.punctuation)
    cov = re.sub(pattern, "", cov).strip().lower()
    cov = int(cov)
    cov = (cov / 1000) * 100

    if cov < THRESHOLD:
        print(
            "Coverage threshold failure: Expected a threshold of {}%, but found {}%"
            .format(THRESHOLD, cov))
        sys.exit(1)
    else:
        print("Current coverage achieved {}%".format(cov))
        sys.exit(0)


if __name__ == "__main__":
    checkCoverage()


example/main.dart
import 'package:flutter/material.dart';
import 'package:shared_themes/colors.dart';
import 'package:shared_themes/spaces.dart';
import 'package:shared_themes/text_themes.dart';

void main() {
  runApp(const SharedThemesExample());
}

class SharedThemesExample extends StatefulWidget {
  /// This widget is the root of your application.
  const SharedThemesExample({Key? key}) : super(key: key);

  @override
  _SharedThemesExampleState createState() => _SharedThemesExampleState();
}

class _SharedThemesExampleState extends State<SharedThemesExample> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Debug Logger example'),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            const Text(
              'You have pushed the button this many times:',
            ),

            /// Uses the sizes exported as doubles in `spaces.dart`
            const SizedBox(height: Sizing.size4),
            /// Use the text themes defined `text_themes.dart`
            Text(
              'Counter value',
              style: TextThemes.normalSize20Text(),
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: () {},
        tooltip: 'Increment',
        /// Uses the color constants exported as colors in `colors.dart`
        backgroundColor: red,
        child: const Icon(Icons.add),
      ),
    );
  }
}


lib/app_theme.dart
import 'package:flutter/material.dart';

class AppTheme {
  static ThemeData getAppTheme(String flavour) {
    return ThemeData(
      fontFamily: 'Lato',
      primaryColor: AppColors.consumerColors['primaryColor'],
      primaryColorDark: AppColors.consumerColors['primaryDark'],
      accentColor: AppColors.consumerColors['accentColor'],
      backgroundColor: AppColors.consumerColors['backgroundColor'],
      textSelectionTheme: TextSelectionThemeData(
        cursorColor: const Color.fromRGBO(66, 133, 244, 1.0),
        selectionColor: AppColors.consumerColors['textSelectionColor'],
        selectionHandleColor: const Color(0xff1de9b6),
      ),
    ).copyWith(
      pageTransitionsTheme: const PageTransitionsTheme(
        builders: <TargetPlatform, PageTransitionsBuilder>{
          TargetPlatform.iOS: FadeUpwardsPageTransitionsBuilder(),
          TargetPlatform.android: FadeUpwardsPageTransitionsBuilder(),
        },
      ),
    );
  }
}

class AppColors {
  AppColors._(); // this basically makes it so you can instantiate this class

  static const Map<String, Color> consumerColors = <String, Color>{
    'primaryColor': Color(0xFF603A8B),
    'primaryDark': Color(0xFF7949AF),
    'accentColor': Color(0xFF739A00),
    'backgroundColor': Color(0xFFF4F6FA),
    'textSelectionColor': Color(0xff0e2153),
  };

  static const Map<String, Color> proColors = <String, Color>{
    // TODO: add pro colors
  };
}


lib/constants.dart
/// These regexp don't belong here. In the next milestone we'll figure out where best to place theme.
// a valid Kenyan phone number regex
final RegExp kenyanPhoneRegExp = RegExp(
  r'''^(?:254|\+254|0)?((7|1)(?:(?:[0-9][0-9])|(?:0[0-8])|(4[0-1]))[0-9]{6})$''',
);

// a valid US phone number regex
final RegExp americanPhoneRegExp =
    RegExp(r'''^(\+)(\d{1,})([(]{1}\d{1,3}[)]){0,}\d{2}\d{3}\d{5}$''');

// a valid UK phone number regex
final RegExp unitedKingdomRegExp = RegExp(
  r'''^(?:(?:\(?(?:0(?:0|11)\)?[\s-]?\(?|\+)44\)?[\s-]?(?:\(?0\)?[\s-]?)?)|(?:\(?0))(?:(?:\d{5}\)?[\s-]?\d{4,5})|(?:\d{4}\)?[\s-]?(?:\d{5}|\d{3}[\s-]?\d{3}))|(?:\d{3}\)?[\s-]?\d{3}[\s-]?\d{3,4})|(?:\d{2}\)?[\s-]?\d{4}[\s-]?\d{4}))(?:[\s-]?(?:x|ext\.?|\#)\d{3,4})?$''',
);

// a generic regex for all international phone numbers. In our case this will suffice to validate Belgium phone numbers and in the future,
// any new country
final RegExp genericInternationalRegExp = RegExp(
  r'''\+(9[976]\d|8[987530]\d|6[987]\d|5[90]\d|42\d|3[875]\d|2[98654321]\d|9[8543210]|8[6421]|6[6543210]|5[87654321]|4[987654310]|3[9643210]|2[70]|7|1)\d{1,14}$''',
);

List<String> supportedIosVersions = <String>[
  '13',
  '14',
  '15',
  '16',
  '17',
  '18'
];

RegExp emailValidator = RegExp(
  r"^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,253}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,253}[a-zA-Z0-9])?)*$",
);

const double kProblemItemHeight = 160.0;

const String kWhatsAppNumber = '+12028569601';
const String kBewellSupportPhoneNumber = '0790360360';

//long snackbar duration used for more important information
const int kLongSnackBarDuration = 10;

//short snackbar duration used for less important information
const int kShortSnackBarDuration = 5;

class UserFeedBackTexts {
  static const String steadyDevice =
      'Please hold your phone steady to take a clear photo';
  static const String selectFileError = 'Error selecting file';
  static const String uploadFileFail = 'Failed to upload file';
  static const String noFileSelected = 'No file selected';
  static const String savingFile = 'Saving file';
  static const List<String> controlLabels = <String>[
    'Select',
    'Camera',
    'Remove'
  ];
  static String selectOrTakeMessage(String name) =>
      'Select or take a photo of $name';
  static String getErrorMessage([String process = '']) {
    if (process.isEmpty) {
      return 'Sorry, an error occurred. Please try again,'
          ' or contact Slade 360 Be.Well Support support'
          ' on $kBewellSupportPhoneNumber';
    } else {
      return 'Sorry, an error occurred while $process.'
          ' Please try again later, or contact Slade 360 Be.Well '
          'Support on $kBewellSupportPhoneNumber';
    }
  }

  static String getSuccessfulAddMessage(String process) {
    return '$process added successfully';
  }

  static String getSuccessfulRetireMessage(String process) {
    return '$process retired successfully';
  }
}

enum VerificationType { phone, email, reset }

// the duration to wait before showing the resend otp widget
const int otpResendTimeoutDuration = 60;
